### PR TITLE
geckodriver: 0.22.0 -> 0.26.0

### LINKED ISSUE #76464: geckodriver update to 0.26
geckodriver is currently stuck at 0.22 because the packaging changed to remove the Cargo.lock file. The recommended way to get the source now is to checkout the mozilla-central megarepo and build the testing/geckodriver folder. 

It doesn't seem like a good idea to clone the whole mozilla megarepo to get the source so perhaps the only way to unstick this issue is to package the binary instead.

See #35301 for some discussion about this already.

Any opinion @das-g ?

### PR DESCRIPTION
**Pull request by arcnmx**
Closes #76464 (see there for some discussion regarding what happened to the source)

###### Motivation for this change

`0.22.0` is well over a year old.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/geckodriver/default.nix [hunk 1]
```diff
@@ -1,26 +1,30 @@
...
-  src = fetchFromGitHub {
-    owner = "mozilla";
-    repo = "geckodriver";
-    rev = "v${version}";
-    sha256 = "12m95lfqwdxs2m5kjh3yrpm9w2li5m8n3fw46a2nkxyfw6c94l4b";
-  };
+  src = let
+    # Source revisions are noted alongside the binary releases:
+    # https://github.com/mozilla/geckodriver/releases
+    rev = "e9783a644016aa9b317887076618425586730d73";
```
Comment (alyssais) [inline pkgs/development/tools/geckodriver/default.nix]: I think it would be cleaner to just inline this.

File: pkgs/development/tools/geckodriver/default.nix [hunk 2]
```diff
@@ -1,26 +1,30 @@
...
+  in (fetchzip {
+    url = "https://hg.mozilla.org/mozilla-central/archive/${rev}.zip/testing";
+    sha256 = "0m86hqyq1jrr49jkc8mnlmx4bdq281hyxhcrrzacyv20nlqwvd8v";
+  }).overrideAttrs (_: {
+    unpackCmd = "unzip $curSrc";
```
Comment (alyssais) [inline pkgs/development/tools/geckodriver/default.nix]: Why is this required?

File: pkgs/development/tools/geckodriver/default.nix (hunk 1 above)
Comment (arcnmx) [reply pkgs/development/tools/geckodriver/default.nix]: Somewhat strongly disagree, but can do.

File: pkgs/development/tools/geckodriver/default.nix (hunk 2 above)
Comment (arcnmx) [reply pkgs/development/tools/geckodriver/default.nix]: There might be a better way to do this? This is here because [unpack phases choose the command based on file extension](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/archivers/unzip/setup-hook.sh) and `fetchzip` as far as I know doesn't provide a way to change the filename.

File: pkgs/development/tools/geckodriver/default.nix (hunk 1 above)
Comment (alyssais) [reply pkgs/development/tools/geckodriver/default.nix]: If you strongly disagree, feel free to leave it as you like. :)

File: pkgs/development/tools/geckodriver/default.nix (hunk 2 above)
Comment (alyssais) [reply pkgs/development/tools/geckodriver/default.nix]: How about

```
unpackPhase = ''
  unzip $src
'';
```

With a comment about why the normal unpackPhase doesn’t work?

Comment (alyssais) [reply pkgs/development/tools/geckodriver/default.nix]: If you prefer it as you have it now that’s also fine.

Comment (arcnmx) [reply pkgs/development/tools/geckodriver/default.nix]: I'm not sure if that will work? `fetchzip` explicitly calls `unpackFile` as part of `postFetch`, but it makes sense to add a comment regarding why it's there.

File: pkgs/development/tools/geckodriver/default.nix (hunk 1 above)
Comment (arcnmx) [reply pkgs/development/tools/geckodriver/default.nix]: *shrug* I'm not invested either way :p

File: pkgs/development/tools/geckodriver/default.nix (hunk 2 above)
Comment (alyssais) [reply pkgs/development/tools/geckodriver/default.nix]: Looks like you’re right. :(
